ESP32 RISCV-C platforms support breaks espressif32 builds

Hi,

the "ESP32 RISCV-C platforms support" commit (e862d5f338afabf21572de7847e56ea81531e00a) unfortunately breaks the espressif32 platform builds:

```
In file included from .pio/libdeps/esp32dev/OneWireNg/src/OneWireNg_CurrentPlatform.h:36:0,
                 from .pio/libdeps/esp32dev/OneWireNg/src/OneWire.h:19,
                 from .pio/libdeps/esp32dev/DallasTemperature/DallasTemperature.h:25,
                 from src/sensors/onewire_temperature.h:6,
                 from src/sensors/onewire_temperature.cpp:1:
.pio/libdeps/esp32dev/OneWireNg/src/platform/OneWireNg_ArduinoESP32.h: In member function 'void OneWireNg_ArduinoESP32::initDtaGpio(unsigned int, bool)':
.pio/libdeps/esp32dev/OneWireNg/src/platform/OneWireNg_ArduinoESP32.h:21:33: error: request for member 'val' in 'GPIO.<anonymous struct>::in', which is of non-class type 'volatile uint32_t {aka volatile unsigned int}'
 # define REG_GPIO_IN_LO GPIO.in.val
                                 ^
```

--- a/src/platform/OneWireNg_ArduinoESP32.h
+++ b/src/platform/OneWireNg_ArduinoESP32.h
@@ -17,7 +17,11 @@
 #include "Arduino.h"
 #include "OneWireNg_BitBang.h"
 
-#if (SOC_GPIO_PIN_COUNT < 32)
+#ifndef GPIO_PIN_COUNT
+# error "GPIO_PIN_COUNT not defined for this version of SDK"
+#endif
+
+#if (GPIO_PIN_COUNT <= 32)
 # define REG_GPIO_IN_LO GPIO.in.val
 # define REG_GPIO_OUT_SET_LO GPIO.out_w1ts.val
 # define REG_GPIO_OUT_CLR_LO GPIO.out_w1tc.val
@@ -160,7 +164,7 @@ protected:
             _dtaGpio.modSetReg = &REG_GPIO_MOD_SET_LO;
             _dtaGpio.modClrReg = &REG_GPIO_MOD_CLR_LO;
         }
-#if (SOC_GPIO_PIN_COUNT >= 32)
+#if (GPIO_PIN_COUNT > 32)
         else {
             _dtaGpio.bmsk = (uint32_t)(1UL << (pin-32));
             _dtaGpio.inReg = &REG_GPIO_IN_HI;
@@ -185,7 +189,7 @@ protected:
             _pwrCtrlGpio.modSetReg = &REG_GPIO_MOD_SET_LO;
             _pwrCtrlGpio.modClrReg = &REG_GPIO_MOD_CLR_LO;
         }
-#if (SOC_GPIO_PIN_COUNT >= 32)
+#if (GPIO_PIN_COUNT > 32)
         else {
             _pwrCtrlGpio.bmsk = (uint32_t)(1UL << (pin-32));
             _pwrCtrlGpio.outSetReg = &REG_GPIO_OUT_SET_HI;

